CMakeLists.txt

```diff
@@ -1,21 +1,77 @@
 cmake_minimum_required(VERSION 3.19)
 cmake_policy(VERSION 3.19)
 
-# Initialize Mbed OS build system. 
-# Note: This block must be before the include of app.cmake
-set(MBED_APP_JSON_PATH mbed_app.json)
-set(CUSTOM_TARGETS_JSON_PATH custom_targets.json) # This activates the custom targets in custom_targets.json
+#### Initialize Mbed OS build system. ####
+######################################################################################################
+### Block of including .json5 files. Files of this block must be included before the app.cmake
 
+#[[ Set path of mbed_app.json (necessary everytime if mbed_app.json5 is in project) ]]
+set(MBED_APP_JSON_PATH mbed_app.json5)
+
+###--------------------------------------------------------------------------------------------------- 
+#[[ This part is intended to custom targets5 and activates the custom_targets.json5 
+and upload method config otherwise functions below should be commented ]]
+
+#[[ Here set path for custom_targets.json5 (this is our default) ]]
+ set(CUSTOM_TARGETS_JSON_PATH custom_targets/custom_targets.json5) 
+
+#[[ Here you can set path for custom upload config .cmake (optional example) ]]
+# set(CUSTOM_UPLOAD_CFG_PATH ${CMAKE_SOURCE_DIR}/${MBED_TARGET}/${MBED_TARGET}.cmake)
+
+#[[ Note: For custom target you need also an upload method and we have few options how you can do that
+    - use the variable CUSTOM_UPLOAD_CFG_PATH below
+    - use default expected path for custom targets upload methods where you create your own config
+      MY_PROJECT/custom_targets/upload_method_cfg
+    - of course you can do it by your self dierectly via cmake in this file
+
+    For regular targets we use mbed-os/tree/master/targets/upload_method_cfg ]]
```

Comment:
Can we link to the upload method wiki page and also to an example upload method config from Mbed CE?

Reply:
I had the link there but was removed before this PR :)

About "of course you can do it by yourself directly via cmake in this file" this is just or info. I saw CMake experienced user use this way automatically.
I suggest to add just the link and the rest into the wiki page. What do you think?